File: Ruby/SuppliesPackage.rb
#encoding:utf-8

module Deepspace

    class SuppliesPackage

        def initialize(_ammoPower,_fuelUnits,_shieldPower)
            @ammoPower = _ammoPower
            @fuelUnits = _fuelUnits
            @shieldPower = _shieldPower
        end

        def self.newCopy(sp)
            new(sp.ammoPower,sp.fuelUnits,sp.shieldPower)
        end

        def ammoPower
            @ammoPower
        end

        def fuelUnits
            @fuelUnits
        end

        def shieldPower
            @shieldPower
        end

        def to_s()
            "This Supplies package gives fuel:" + @fuel + ", ammoPower:" + @ammoPower + ", shieldPower:" + @shieldPower
        end
        

        
    end

end

File: Java_Exam_P123/deepspace/src/main/java/deepspace/SuppliesPackage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author fjaviersaezm
 */
class SuppliesPackage {
 
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
   

    SuppliesPackage(float ammoPower, float fuelUnits, float shieldPower) {
        this.ammoPower = ammoPower;
        this.fuelUnits = fuelUnits;
        this.shieldPower = shieldPower;
    }

    SuppliesPackage(SuppliesPackage p){
        this.ammoPower = p.getAmmoPower();
        this.fuelUnits = p.getFuelUnits();
        this.shieldPower = p.getFuelUnits();
    }
    
    public float getAmmoPower() {
        return ammoPower;
    }

    public float getFuelUnits() {
        return fuelUnits;
    }

    public float getShieldPower() {
        return shieldPower;
    }
    
    public String toString(){
        String s = String.format("This Supplies package has: ammoPower = %f , fuelUnits = %f , shieldPower = %f .", ammoPower, fuelUnits , shieldPower);
        return s;
    }
    
    
}
